Aptoide/aptoide-client-v8
# [ASV-1087] Sequential Downloads Implementation

**What does this PR do?**
One of the downloads requirement is to have a queue in order to avoid concurrency problems.
This PR corresponds to the implementation of a queue, where it is checked if there is any currently active download. If there isn't, then it will check the downloads IN_QUEUE and take the first one.

**Database changed?**

   Yes (DownloadsAccessor)

**Where should the reviewer start?**

- [ ] AptoideDownloadManager.java

**How should this be manually tested?**

Since this aims at implementing a queue of downloads, then it is easier to test it on the Apps bottom navigation section where it is faster to start several downloads and check the progress of all of them at the same time.

PS: None of the downloads will start the android install app popup dialog, since that is already being taken care of in another ticket ([ASV-1089](https://aptoide.atlassian.net/browse/ASV-1089))

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-1087](https://aptoide.atlassian.net/browse/ASV-1087)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.) 

No


**Code Review Checklist**
